Singer.API/ClientApp/src/app/modules/admin/components/add-family-wizard/add-family-wizard.component.ts

```diff
@@ -0,0 +1,312 @@
+               res => {
+                  //Update the local instance
+                  let legalGuardianIndex = this.legalGuardians.findIndex(guardian => guardian.id == result.id);
+                  this.legalGuardians[legalGuardianIndex] = result;
+
+
+                  this.snackBar.open(
+                     `${result.firstName} ${result.lastName} werd aangepast.`,
+                     'OK',
+                     { duration: 2000 }
+                  );
+               },
+               err => {
+                  this.handleApiError(err);
+               }
+            );
+         }
+      );
+   }
+
+   deleteLegalGuardian(legalGuardian: LegalGuardian): void {}
+
+   addCareUser(): void {
+      const dialogRef = this.dialog.open(CareUserDetailsComponent, {
+         data: {
+            careUserInstance: null,
+            isAdding: true,
+            displayContactFields: false,
+         },
+         width: '80vw',
+      });
+
+      dialogRef.componentInstance.submitEvent.subscribe((result: CareUser) => {
+         this.careUserService.createCareUser(result).subscribe(
+            res => {
+               //Save result localy for linking users
+               result.id = res.id;
+               this.careUsers.push(result);
+
+               this.snackBar.open(
+                  `Gebruiker ${result.firstName} ${result.lastName} werd toegevoegd.`,
+                  'OK',
+                  { duration: 2000 }
+               );
+            },
+            err => {
+               this.handleApiError(err);
+            }
+         );
+      });
+   }
+
+   editCareUser(careUser: CareUser): void {
+      //Dereference careUser to avoid updating local instance when API might refuse the update
+      const deRefCareUser = { ...careUser };
+      const dialogRef = this.dialog.open(CareUserDetailsComponent, {
+         data: {
+            careUserInstance: deRefCareUser,
+            isAdding: false,
+            displayContactFields: false,
+         },
+         width: '80vw',
+      });
+
+      dialogRef.componentInstance.submitEvent.subscribe((result: CareUser) => {
+         // Update the Careuser
+         this.careUserService.updateUser(result).subscribe(
+            res => {
+               //Update the local instance
+               let careUserIndex = this.careUsers.findIndex(user => user.id == result.id);
+               this.careUsers[careUserIndex] = result;
+
+               this.snackBar.open(
+                  `Gebruiker ${result.firstName} ${result.lastName} werd aangepast.`,
+                  'OK',
+                  { duration: 2000 }
+               );
+            },
+            err => {
+               this.handleApiError(err);
+            }
+         );
+      });
+   }
+
+   deleteCareUser(careUser: CareUser): void {}
+
+   linkUsers(): boolean {
+      //collect all the careUser id's
+      var careUsersToAdd: string[] = [];
+      this.careUsers.forEach(careUser => {
```

Comment:
Why don't you just do the following?
`const careUsersToAdd = this.careUsers.map(careUser => careUser.id);`
?

Reply:
I didn't know that was an option😅 It's fixed now.